### PR TITLE
Fix tests and replace sockets extension with built-in streams API

### PR DESCRIPTION
**Pull request by aldas**
This PR contains a lot of things:

* I fixed/replaced all old tests
* reformated code
* replaced dependency to php sockets extension (not enabled by default) with built-in Streams API
* refactored a lot of stuff to specific classes - that ModbusMaster.php had just to many things inside. Also I thought that instead of plain stream or sockets maybe I could use in my project [reactphp/socket-client](https://github.com/reactphp/socket-client) as a underlying client so having stuff releated to modbus functions and packets and network releated already somewhat separated would make it easier for my private use.
* added support for Travis CI (https://travis-ci.org/aldas/phpmodbus/builds this is view to my dev fork). I recomend you to use travis for master branch also after pr has been merged. This is free for open source projects. Just link you github account to it and enable Travis for this repo. NB: add "MOCKSERVER_TIMEOUT_USEC" env variable with value "1000000" so tests will not be flaky.

I did not change library API but I did change minimal PHP version from 5.3.2 to 5.5 as I used features introduced in that version (finally block if I recall correctly).

Nb: tests may be flaky if host is under heavy load - mock server subprocess warm up takes time.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+cache:
+    directories:
+        - $HOME/.composer/cache
+
+php:
+    - 5.6
+    - 7.0
+
+branches:
```
Comment (aldas) [inline .travis.yml:15]: I think this "branches" part should be removed if tests are merged to master.

**Comment by MightyPork** [comment]
I'll have to read through it when I have more time, but at first glance, a lot of changes I don't really like.

tabs->spaces. A matter of preference, I hate spaces. But ok...

The `sendAndReceive` and splitting packets to separate classes looks like obfuscation to me, to be honest. They don't even share a common interface, it's just "oh this class is too long, let's split it somehow". If I compare it to the old code, it looks way more confusing. One alternative could be to split the Master class to traits for the different packets, but I'm not sure about that either, sounds kind of dirty.

Then there's the thing with the Socket class, which has the same fields as the Master class (also each command creates a new socket, I know it was this way in the old code too, but maybe we could re-use it?).

Perhaps at this point abandoning the old API and making it all clean and well structured, without all this baggage, would be better. I don't know.. but I'm hesitant to merge it in it's current form.

**Comment by aldas** [comment]
* regarding tabs vs spaces - [PSR-2](http://www.php-fig.org/psr/psr-2/) 

> Code MUST use 4 spaces for indenting, not tabs.

* I think that having smaller classes with specific responsibility is better than having one gigantic class being responsible for everything. Having packet related stuff in traits may be cleaner (I'm no php developer but more of a Java developer by experience). 
I think If new API is to be built and API backwards compatibility is maintained these public properties will not go anywhere but otherwise this ModbusMaster could be turned to a "adapter" for the new API. Just move everything releated to new API to another subnamespace so new and old will not be intermixed. And may even not hurt eyes that bad (just annotate as deprecated and we are good to go)
There are weird methods with old API - ala `writeSingleCoil` takes data as array that could be improved in new API.

* regarding new socket been open for every modbus function call - I think there is a fork project where this has been redone https://github.com/tobiaskluge/phpmodbus

**Comment by aldas** [comment]
If you happen to have time please check PRs code and provide list of things that should be fixed/changed to get this merged. I think having ext_sockets dropped with still some of the old mess with a little new is better than not moving forward. As for new api. I think it should deserve its own separate issue. I can work on it as I have some ideas for functionality I would want for my own project but adding it to old api does not sound so nice idea (automatically splitting to multiple requests if quantity is too big, I think that [non-blocking streams can process multiple requests in parallel](http://stackoverflow.com/a/11705584/2514290))

on top of my head I know that I should fix:
* go back to supporting php 5.4- code/tests
* ModbusSocketBuilder extending ModbusSocket is a dirty way to access protected member as you could access methods not meant for builder

**Comment by MightyPork** [comment]
I don't think there's any point in supporting 5.4, 5.5 maybe, but it's already EOL as well.

My aim with this project was really to just add namespace, re-format to tabs (screw PSR-2, really), and clean it a bit, so it can be used in Composer. 

Maybe it'll be better if you continue the development in your fork, for now - I don't see having much time to mess with this. If you figure out some shiny new API, it's better than trying to polish this mess

**Comment by aldas** [comment]
I guess you are right. I'll work on my idea further in my fork. I'll close this PR

**Pull request closed without merging**